tests/shouldwork/DDR/VendorDDR.hs

```diff
@@ -0,0 +1,335 @@
+  (clkSlow, clkFast, rstSlow, enSlow, inIn, outIn, done)
+ where
+  (genInOut1, genInOut2, vendorInOut1, vendorInOut2) = unbundle inOut
+  (genOutOut, vendorOutOut) = unbundle outOut
+  inIn = cToDFast cntr4
+  outIn = bundle (posIn, negIn)
+  posIn = cToDSlow $ cntr4 + 402
+  negIn = cToDSlow $ cntr4 + 802
+  cToDFast = unsafeSynchronizer clk4 clkFast . fmap (truncateB . (`shiftR` 1))
+  cToDSlow = unsafeSynchronizer clk4 clkSlow . fmap (truncateB . (`shiftR` 2))
+  cntr4 :: Signal Dom4 C
+  cntr4 = register clk4 noReset enableGen 0 $ cntr4 + 1
+#ifndef INTEL_VERILOG
+  done1 = outputVerifierWith (\clk rst -> assert clk rst "genOutOut")
+    clkFast clkFast noReset expGenOut $
+      ignoreFor clkFast noReset enableGen d1 0 genOutOut
+#else
+  -- The test contains a number of (implicit) parallel, coinciding processes.
+  -- Execution for these processes is left undefined in the Verilog spec, nor is
+  -- there a mechanism to get consistent behavior (like in VHDL). Therefore,
+  -- simulators are free to pick any execution order they'd like. With Verilog
+  -- HDL, ModelSim executes the processes in an order that makes this particular
+  -- `outputVerifier` fail. SystemVerilog however is unaffected, as of course is
+  -- VHDL.
+  --
+  -- CI never runs @IntelDDR.hs@ (which imports this module), but to enable
+  -- manual verification with @IntelDDR.hs@, you can define the macro
+  -- INTEL_VERILOG when compiling. That way, you can disable this
+  -- `outputVerifier` so you can still verify the vendor primitives, which is
+  -- the main purpose of the test.
+  --
+  -- Note that @IntelDDR.hs@ still suffers from the issue described in
+  -- https://github.com/clash-lang/clash-compiler/issues/2854
+  --
+  -- The `const` is just there to prevent a @-Wunused-matches@ on @expGenOut@
+  -- and @genOutOut@.
+  done1 = pure $ const True (expGenOut, genOutOut)
+#endif
+  done2 = outputVerifierWith (\clk rst -> assert clk rst "vendorOutOut")
+    clkFast clkFast noReset expVendorOut vendorOutOut
+  done3 = outputVerifierWith (\clk rst -> assert clk rst "genInOut1")
+    clkSlow clkSlow noReset (map fst expIn0) $
+      ignoreFor clkSlow noReset enableGen d2 0 genInOut1
+  done4 = outputVerifierWith (\clk rst -> assert clk rst "genInOut2")
+    clkSlow clkSlow noReset (map snd expIn0) $
+      ignoreFor clkSlow noReset enableGen d1 0 genInOut2
+  done5 = outputVerifierWith (\clk rst -> assert clk rst "vendorInOut1")
+    clkSlow clkSlow noReset (map fst expIn0) $
+      ignoreFor clkSlow noReset enableGen d2 0 vendorInOut1
+  done6 = outputVerifierWith (\clk rst -> assert clk rst "vendorInOut2")
+    clkSlow clkSlow noReset (map snd expIn0) vendorInOut2
+  doneFast = done1 `strictAnd` done2
+  doneSlow = done3 `strictAnd` done4 `strictAnd` done5 `strictAnd` done6
+  done = doneFast `strictAnd` unsafeSynchronizer clkSlow clkFast doneSlow
+  notDone = not <$> done
+  clk4 = tbClockGen $ unsafeSynchronizer clkFast clk4 notDone
+  -- {-# NOINLINE clk4 #-}
```

Comment:
Ah these can still be removed. They were part of an experiment to see if I could improve HDL generation times.